openmdao/solvers/solver.py

```diff
@@ -591,13 +592,30 @@ def _solve(self):
         self._mpi_print(self._iter_count, norm, norm / norm0)
 
         stalled = False
+        stall_count = 0
         if stall_limit > 0:
-            stall_count = 0
             stall_norm = norm0
 
         while self._iter_count < maxiter and norm > atol and norm / norm0 > rtol and not stalled:
             with Recording(type(self).__name__, self._iter_count, self) as rec:
-                self._single_iteration()
+
+                if stall_count == 3 and not self.linesearch.options['print_bound_enforce']:
+
+                    self.linesearch.options['print_bound_enforce'] = True
+
+                    msg = ("Your model has stalled three times and may be violating the "
+                           "bounds. In the future, turn on print_bound_enforce in your "
+                           "solver options to see. For example:\n"
+                           "'prob.model.nonlinear_solver.linesearch.options"
+                           "['print_bound_enforce']=True'. \nThe bound(s) being violated "
+                           "now are:\n")
+                    simple_warning(msg)
+
+                    self._single_iteration()
+                    self.linesearch.options['print_bound_enforce'] = False
```

Comment:
Warning messages are normally written to `stderr` or handled by `logging` while the bound violations are being printed to `stdout` (see  [_print_violations](https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/solvers/linesearch/backtracking.py#L17)).  This raises the possibility that this output is not seen as intended.

- Maybe the proper thing to do is to make the bounds violation messages proper warnings instead of just prints? 
- Could also try to capture `stdout` and insert the violation info into the warning message, but there's probably a bunch of other output from that `_single_iteration`

Reply:
yeah, I think the bounds violation prints would be better as warnings (simple_warning) rather than prints (though maybe verify with Jubs).